src/client.ts

```diff
@@ -101,6 +100,7 @@ export class Client {
   config: any;
   streamId: any;
 
+  http: HttpClient;
```

Comment:
Moved all http request logic into a separate class so retry state can be manage easier.